deployments/core.go

```diff
@@ -0,0 +1,255 @@
+	}
+
+	message := "Deleting Core component namespace " + coreDeploymentNamespace
+	_, err = helpers.WaitForCommandCompletion(ui, message,
+		func() (string, error) {
+			return "", c.DeleteNamespace(coreDeploymentNamespace)
+		},
+	)
+	if err != nil {
+		return errors.Wrapf(err, "Failed deleting namespace %s", coreDeploymentNamespace)
+	}
+
+	message = "Waiting for Core namespace to be gone"
+	warning, err := helpers.WaitForCommandCompletion(ui, message,
+		func() (string, error) {
+			var err error
+			for err == nil {
+				_, err = c.Kubectl.CoreV1().Namespaces().Get(
+					context.Background(),
+					coreDeploymentNamespace,
+					metav1.GetOptions{},
+				)
+			}
+			if serr, ok := err.(*apierrors.StatusError); ok {
+				if serr.ErrStatus.Reason == metav1.StatusReasonNotFound {
+					return "", nil
+				}
+			}
+
+			return "", err
+		},
+	)
+	if err != nil {
+		return err
+	}
+	if warning != "" {
+		ui.Exclamation().Msg(warning)
+	}
+
+	ui.Success().Msg("Core component removed")
+
+	return nil
+}
+
+// Create kubernetes namespace for core component
+func (core Core) createNamespace(c *kubernetes.Cluster, ui *ui.UI) error {
+	if _, err := c.Kubectl.CoreV1().Namespaces().Create(
+		context.Background(),
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: coreDeploymentNamespace,
+				Labels: map[string]string{
+					"quarks.cloudfoundry.org/monitored": "quarks-secret",
+					kubernetes.FusemlDeploymentLabelKey: kubernetes.FusemlDeploymentLabelValue,
+				},
+			},
+		},
+		metav1.CreateOptions{},
+	); err != nil {
+		return nil
+	}
+
+	if err := c.LabelNamespace(coreDeploymentNamespace, kubernetes.FusemlDeploymentLabelKey, kubernetes.FusemlDeploymentLabelValue); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Create secret to store gitea credentials
+// This might be different gitea instance than the one deployed by fuseml-installer!
+func (core Core) createCoreCredsSecret(c *kubernetes.Cluster, domain string) error {
+
+	// TODO if gitea information were not provided, read them from gitea deployment in this cluster?
+	giteaUsername, exists := os.LookupEnv("GITEA_USERNAME")
+	if !exists {
+		//return errors.New("Value for gitea user name (GITEA_USERNAME) was not provided.")
+		giteaUsername = "dev"
+	}
+	giteaPassword, exists := os.LookupEnv("GITEA_PASSWORD")
+	if !exists {
+		// return errors.New("Value for gitea user password (GITEA_PASSWORD) was not provided.")
+		giteaPassword = "changeme"
+	}
+
+	giteaURL, exists := os.LookupEnv("GITEA_URL")
+	if !exists {
+		// return errors.New("Value for gitea URL (GITEA_URL) was not provided")
+		giteaURL = "http://gitea." + domain
+	}
+	_, err := c.Kubectl.CoreV1().Secrets(coreDeploymentNamespace).Create(context.Background(),
+		&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: coreSecretName,
+			},
+			StringData: map[string]string{
+				"GITEA_USERNAME": giteaUsername,
+				"GITEA_PASSWORD": giteaPassword,
+				"GITEA_URL":      giteaURL,
```

Comment:
Minor detail, but I wouldn't save the URL as a secret, because it's not sensitive data. Supplying it directly as a deployment environment variable value provides better visibility into what's going on for someone who's debugging fuseml-core.

Reply:
OK, good point.